### PR TITLE
[FLINK-14486][table-api, docs] Update documentation regarding Temporary Objects

### PR DESCRIPTION
**Pull request by dawidwys**
## What is the purpose of the change

Update documentation with temporary objects. Changed references of `scan/registerTable/registerTableSource/registerTableSink` to `from/createTemporaryView/createTemporaryTable` wherever possible.

Added section describing differences between temporary tables and permanent tables.

Added description of what a virtual table (VIEW) and table(TABLE) are.

Added description of path expanding with catalog and database

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (not applicable / **docs** / JavaDocs / not documented)

### CODE REVIEW

**Comment by dawidwys** [comment]
@fhueske @sjwiesman Could you have a look at the PR and check if it makes sense?

File: docs/dev/table/common.md [hunk 1]
```diff
@@ -292,25 +292,39 @@ b_b_t_env = BatchTableEnvironment.create(environment_settings=b_b_settings)
...
-* an existing `Table` object, usually the result of a Table API or SQL query.
-* a `TableSource`, which accesses external data, such as a file, database, or messaging system. 
-* a `DataStream` or `DataSet` from a DataStream (only for stream job) or DataSet (only for batch job translated from old planner) program. Registering a `DataStream` or `DataSet` is discussed in the [Integration with DataStream and DataSet API](#integration-with-datastream-and-dataset-api) section.
+On the other hand temporary tables are always stored in memory and exist only for a single session/job.
+Temporary tables are not bound to any catalog or database. They can be created in a namespace of a catalog and database
+that does not exist. They are also not dropped if a corresponding database is removed.
+Moreover temporary tables can shadow permanent tables. It is possible to register a temporary table
+with exactly the same identifier as an existing permanent table. Such a permanent table becomes
+inaccessible as long as the temporary table exists. All queries with that identifier will be executed
+against the temporary table.
```
Comment (sjwiesman) [inline docs/dev/table/common.md]: I think this is confusing and maybe touches on to many implementation details. What about: 

> Tables may either be temporary, and tied to the lifecycle of a single Flink session, or permanent, and visible across multiple Flink sessions and clusters.
> 
> Permanent tables require a [catalog]({{ site.baseurl }}/dev/table/catalogs.html) (such as Hive) to maintain metadata about the table. Once a permanent table is created, it is visible to any Flink session that can connect to the catalog and will continue to exist until the table is explicitly dropped. 
> 
> On the other hand, temporary tables are always stored in memory and only exist for the duration of the Flink session they are created within. These tables are not visible to other clusters. They are not bound to any catalog or database but can be created in the namespace of one. Temporary tables are not dropped if their corresponding database is removed.
> 
> #### Shadowing
> 
> It is possible to register a temporary table with the same identifier as an existing permanent table. The temporary table _shadows_ the permanent one and make the permanent table inaccessible as long as the temporary one exists. All queries with that identifier will be executed against the temporary table.

File: docs/dev/table/common.md [hunk 2]
```diff
@@ -345,133 +359,139 @@ tableEnv.registerTable("projectedTable", projTable)
...
+proj_table = table_env.from_path("X").select(...)
 
 # register the Table projTable as table "projectedTable"
 table_env.register_table("projectedTable", proj_table)
 {% endhighlight %}
 </div>
 </div>
 
-**Note:** A registered `Table` is treated similarly to a `VIEW` as known from relational database systems, i.e., the query that defines the `Table` is not optimized but will be inlined when another query references the registered `Table`. If multiple queries reference the same registered `Table`, it will be inlined for each referencing query and executed multiple times, i.e., the result of the registered `Table` will *not* be shared.
+**Note:** A `Table` object similarly to a `VIEW` from relational database
```
Comment (sjwiesman) [inline docs/dev/table/common.md]:
```suggestion
**Note:** `Table` objects are similar to `VIEW`'s from relational database
```

File: docs/dev/table/common.md [hunk 3]
```diff
@@ -345,133 +359,139 @@ tableEnv.registerTable("projectedTable", projTable)
...
 
-### Register a TableSource
+#### Connector tables
 
-A `TableSource` provides access to external data which is stored in a storage system such as a database (MySQL, HBase, ...), a file with a specific encoding (CSV, Apache \[Parquet, Avro, ORC\], ...), or a messaging system (Apache Kafka, RabbitMQ, ...). 
-
-Flink aims to provide TableSources for common data formats and storage systems. Please have a look at the [Table Sources and Sinks]({{ site.baseurl }}/dev/table/sourceSinks.html) page for a list of supported TableSources and instructions for how to build a custom `TableSource`.
-
-A `TableSource` is registered in a `TableEnvironment` as follows:
+It is also possible to create a `TABLE` as known from relational databases from a [connector]({{ site.baseurl }}/dev/table/connect.html) declaration.
```
Comment (sjwiesman) [inline docs/dev/table/common.md]:
```suggestion
It is also possible to create a `Table` as known from relational databases from a [connector]({{ site.baseurl }}/dev/table/connect.html) declaration.
```

?

File: docs/dev/table/common.md [hunk 4]
```diff
@@ -345,133 +359,139 @@ tableEnv.registerTable("projectedTable", projTable)
...
+tableEnvironment.sqlUpdate("CREATE [TEMPORARY] TABLE MyTable (...) WITH (...)")
 {% endhighlight %}
 </div>
 </div>
 
-**Note:** A `TableEnvironment` used for Blink planner only accepts `StreamTableSource`, `LookupableTableSource` and `InputFormatTableSource`, and a `StreamTableSource` used for Blink planner on batch must be bounded.
+### Expanding table identifiers
 
-{% top %}
+Tables are always registered with a 3 part identifiers consisting of catalog name, database name and
```
Comment (sjwiesman) [inline docs/dev/table/common.md]:
```suggestion
Tables are always registered with a 3 part identifier consisting of catalog, database, and
```

File: docs/dev/table/common.md [hunk 5]
```diff
@@ -345,133 +359,139 @@ tableEnv.registerTable("projectedTable", projTable)
...
+table name. The first two parts are optional. If they are not provided the current default values will
```
Comment (sjwiesman) [inline docs/dev/table/common.md]:
```suggestion
table name. The first two parts are optional and if they are not provided the set default values will
```

File: docs/dev/table/common.md [hunk 6]
```diff
@@ -345,133 +359,139 @@ tableEnv.registerTable("projectedTable", projTable)
...
+be used. Identifiers follow SQL requirements. That means that they can be escaped with ``` character.
```
Comment (sjwiesman) [inline docs/dev/table/common.md]:
```suggestion
be used. Identifiers follow SQL requirements which means that they can be escaped with ``` character.
```

File: docs/dev/table/common.md (hunk 1 above)
Comment (sjwiesman) [reply docs/dev/table/common.md]: I realize the sentence about permanent tables needing to be serializable is likely referring to registered DataStream / DataSet tables. The issue is it feels like explaining why without saying what. I think it makes more sense to leave that out of this section. Instead, in the section on going from DataStream / Set -> Table just say the resulting table is always temporary.

**Comment by sjwiesman** [comment]
Since you dropped all references to `Table#scan` (which I think is fine) it is really important to make sure the new from clause is well documented in the release notes.

File: docs/dev/table/common.md (hunk 3 above)
Comment (dawidwys) [reply docs/dev/table/common.md]: After second thought, I think it is more consistent to leave it with capital letters. In other places, whenever we refer to SQL terms we use capital letters.

File: docs/dev/table/connect.zh.md [hunk 7]
```diff
@@ -110,7 +110,7 @@ table_environment \
     .with_format(...) \
     .with_schema(...) \
     .in_append_mode() \
-    .register_table_source("MyTable")
+    .createTemporaryTable("MyTable")
```
Comment (fhueske) [inline docs/dev/table/connect.zh.md]: should be `create_temporary_table`?

**Comment by fhueske** [comment]
Thanks for the update @dawidwys!
+1 for merging

Best, Fabian

**Pull request closed without merging**